src/wallet/rpcwallet.cpp

```diff
@@ -2931,8 +2932,14 @@ UniValue z_getnewaddress(const UniValue& params, bool fHelp)
         addrType = params[0].get_str();
     }
 
-    KeyIO keyIO(Params());
+    const auto& chainparams = Params();
+    KeyIO keyIO(chainparams);
     if (addrType == ADDR_TYPE_SPROUT) {
+        int nextBlockHeight = chainActive.Height() + 1;
+        if (chainparams.GetConsensus().NetworkUpgradeActive(nextBlockHeight, Consensus::UPGRADE_CANOPY)) {
```

Comment:
This allows generating sprout addresses during the initial block download.

Change this to throw the error if the node is in the initial block download *or* canopy is activated.